### PR TITLE
feat: use filters to export & fix accents issue

### PR DESCRIPTION
**Pull request by Wamibee**

### CODE REVIEW

**Review by Arty3P — DISMISSED** [review]

File: src/main/java/org/example/volunteers/Cleaner.java [hunk 1]
```diff
@@ -34,19 +44,39 @@ public static String capitalizeWord(String str){
...
+        name = name.trim();
+        if(name != null && !name.isEmpty()){
+            name = capitalizeWord(name);
+            name = name.replaceAll(" ", "-");
+            // Replace all characters except a to z, A to Z and dashes for compound noun
+            name = name.replaceAll("[^a-zA-Z-–—−\u00C0-\u00FF]","");
+            return name.substring(0, 1).toUpperCase() + name.substring(1);
+        } else {
+            return "";
+        }
```
Comment (neerfix) [inline src/main/java/org/example/volunteers/Cleaner.java]:
```suggestion
        if(name == null && name.isEmpty()){
            return "";
        }
        
        name = capitalizeWord(name);
            name = name.replaceAll(" ", "-");
            // Replace all characters except a to z, A to Z and dashes for compound noun
            name = name.replaceAll("[^a-zA-Z-–—−\u00C0-\u00FF]","");
            return name.substring(0, 1).toUpperCase() + name.substring(1);
```

**Review by neerfix — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/volunteers/Cleaner.java [hunk 2]
```diff
@@ -34,19 +44,39 @@ public static String capitalizeWord(String str){
...
+        name = name.trim();
+        if(name != null && !name.isEmpty()) {
+            name = capitalizeWord(name);
+            name = name.toLowerCase().replaceAll(" ", "-");
+            // Replace all characters except a to z, A to Z and dashes for compound noun
+            name = name.replaceAll("[^a-zA-Z-–—−\u00C0-\u00FF]", "");
+            return name.toUpperCase();
+        } else {
+            return "";
+        }
```
Comment (neerfix) [inline src/main/java/org/example/volunteers/Cleaner.java]:
```suggestion
        name = name.trim();
        if(name == null && name.isEmpty()) {
            return "";
        }

        name = capitalizeWord(name);
            name = name.toLowerCase().replaceAll(" ", "-");
            // Replace all characters except a to z, A to Z and dashes for compound noun
            name = name.replaceAll("[^a-zA-Z-–—−\u00C0-\u00FF]", "");
            return name.toUpperCase();
```

File: src/main/java/org/example/volunteers/Cleaner.java [hunk 3]
```diff
@@ -34,19 +44,39 @@ public static String capitalizeWord(String str){
...
+    }
+
+    public static String formatNickName(String nickname){
+        nickname = nickname.trim();
+        if(nickname != null && !nickname.isEmpty()) {
+            nickname = nickname.replaceAll(" ", "").replaceAll("[^a-zA-Z0-9\u00C0-\u00FF-_&@€!]", "");
+            return nickname;
+        } else {
+            return "";
+        }
```
Comment (neerfix) [inline src/main/java/org/example/volunteers/Cleaner.java]:
```suggestion
        nickname = nickname.trim();
        if(nickname == null && nickname.isEmpty()) {
            return "";
        }
        
         nickname = nickname.replaceAll(" ", "").replaceAll("[^a-zA-Z0-9\u00C0-\u00FF-_&@€!]", "");
            return nickname;

```

File: src/main/java/org/example/volunteers/Cleaner.java [hunk 4]
```diff
@@ -81,13 +111,18 @@ public static String cleanPhoneNumber(String phoneNumber) {
...
+            java.util.regex.Pattern p = java.util.regex.Pattern.compile(ePattern);
+            java.util.regex.Matcher m = p.matcher(email);
+            if(m.matches()){
+                return email.toLowerCase();
+            } else {
+                return "";
+            }
         } else {
             return "";
         }
```
Comment (neerfix) [inline src/main/java/org/example/volunteers/Cleaner.java:128]: Pareil ici, flemme de faire des suggestions

**Review by neerfix — CHANGES_REQUESTED** [review]

**Review by neerfix — DISMISSED** [review]

**Review by neerfix — APPROVED** [review]

**Pull request merged**